client/dashboard/sites/staging-site-sync-modal/index.tsx

```diff
@@ -0,0 +1,275 @@
+			),
+			syncFrom: 'production',
+			syncTo: 'staging',
+		},
+		fromLabel: __( 'Push' ),
+		toLabel: __( 'To' ),
+		learnMore: __( 'Read more about <a>environment push</a>.' ),
+		submit: __( 'Push' ),
+	};
+};
+
+export default function StagingSiteSyncModal( {
+	onClose,
+	syncType,
+	environment,
+	productionSiteId,
+	stagingSiteId,
+}: StagingSiteSyncModalProps ) {
+	const syncConfig = getSyncConfig( syncType );
+	const [ domainConfirmation, setDomainConfirmation ] = useState( '' );
+
+	const targetEnvironment = syncConfig[ environment ].syncTo;
+	const sourceEnvironment = syncConfig[ environment ].syncFrom;
+
+	const { data: productionSite } = useQuery( {
+		...siteByIdQuery( productionSiteId ?? 0 ),
+		enabled: !! productionSiteId,
+	} );
+	const productionSiteSlug = productionSite?.slug;
+	const productionSiteTitle = productionSite?.name;
+
+	const { data: stagingSite } = useQuery( {
+		...siteByIdQuery( stagingSiteId ?? 0 ),
+		enabled: !! stagingSiteId,
+	} );
+	const stagingSiteSlug = stagingSite?.slug;
+	const stagingSiteTitle = stagingSite?.name;
+
+	const targetSiteSlug = targetEnvironment === 'production' ? productionSiteSlug : stagingSiteSlug;
+
+	const sourceSiteTitle = sourceEnvironment === 'staging' ? stagingSiteTitle : productionSiteTitle;
+	const targetSiteTitle =
+		targetEnvironment === 'production' ? productionSiteTitle : stagingSiteTitle;
+
+	// TODO: Uncomment once we need it.
+	// const querySiteId = sourceEnvironment === 'staging' ? stagingSiteId : productionSiteId;
+
+	const handleClose = useCallback( () => {
+		onClose();
+	}, [ onClose ] );
+
+	const handleConfirm = () => {
+		// Sync everything
+		// TODO: Uncomment once pullFromStaging and pushToStaging are available
+		// const include_paths = '';
+		// const exclude_paths = '';
+
+		if (
+			( syncType === 'pull' && environment === 'production' ) ||
+			( syncType === 'push' && environment === 'staging' )
+		) {
+			// TODO: Replace with the correct pullFromStaging (once it is moved to V2)
+			// pullFromStaging( { types: 'paths', include_paths, exclude_paths } );
+		} else {
+			// TODO: Replace with the correct pushToStaging (once it is moved to V2)
+			// pushToStaging( { types: 'paths', include_paths, exclude_paths } );
+		}
+	};
+
+	const handleDomainConfirmation = useCallback(
+		( value: string | undefined ) => setDomainConfirmation( value || '' ),
+		[]
+	);
+
+	const showDomainConfirmation = targetEnvironment === 'production';
+
+	const isSubmitDisabled = showDomainConfirmation && domainConfirmation !== productionSiteSlug;
+
+	return (
+		<Modal
+			title={ syncConfig[ environment ].title }
+			onRequestClose={ handleClose }
+			style={ { maxWidth: '668px' } }
+		>
+			<VStack spacing={ 5 }>
+				<Text>
+					{ createInterpolateElement( syncConfig[ environment ].description, {
+						a: <ExternalLink href={ `/backup/${ targetSiteSlug }` } children={ null } />,
```

Comment:
Nit: the link text is "Activity Log" here but we're redirecting to the backup page. We could update this to point to `/activity-log/${ targetSiteSlug }` to show this page in the WP admin instead. 

<img width="1221" height="939" alt="Image" src="https://github.com/user-attachments/assets/98ceba08-3e1f-4a3a-a82b-6afddcc4fb3e" />

| backup page | activity log page |
|--------|--------|
| <img alt="Image" src="https://github.com/user-attachments/assets/df412576-30b6-4641-8d88-0b19494e4b88" /> | <img alt="Image" src="https://github.com/user-attachments/assets/8f1dfe41-2a7c-4790-99d1-14f5506e7184" /> |

Reply:
That's a good catch! I have fixed it in https://github.com/Automattic/wp-calypso/pull/105352/commits/5bea90279a990d0fdb4e6b555ea3e45d1c6e9d44 and applied the same fix for the Sync modal in V1 as well (https://github.com/Automattic/wp-calypso/pull/105352/commits/0ee0cdd8299d8fc4bdeb8a618f570a3b80630638).